### DISCUSSION TITLE
Investigate on ICS29 fee middleware

### DISCUSSION THREAD

**Discussion by thomas-nguy** [crypto-org-chain/cronos#237] [Ideas]
In order to bring cosmos tokens to cronos, we need a fee mechanism to cover the cost of maintaining relayers 

ICS29 fee middleware in ibc-go seems still WIP, but worth to have a look and experiment

https://github.com/cosmos/ibc-go/tree/ics29-fee-middleware

spec
https://github.com/cosmos/ibc/tree/master/spec/app/ics-029-fee-payment

**Comment by adu-web3** [comment]
I can work on investigating the ICS29 and how could I experiment with this? @thomas-nguy

**Comment by thomas-nguy** [comment]
Maybe the first task will be to look at the repo, try to understand how does it work and how far are we from completion by reading the integration/unit tests

Then try to setup a local instance with the module

**Reply by adu-crypto** [reply]
okay.

**Reply by thomas-nguy** [reply]
I have updated the description with the spec link

**Reply by adu-crypto** [reply]
ok. I have been reading though this spec and now need to review the code.
